stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -66,10 +78,17 @@ struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
     recursive_reducer(recursive_reducer& other, tbb::split)
         : num_vars_per_term_(other.num_vars_per_term_),
           num_vars_shared_terms_(other.num_vars_shared_terms_),
-          sliced_partials_(other.sliced_partials_),
+          varis_(other.varis_),
+          partials_(other.partials_),
           vmapped_(other.vmapped_),
           msgs_(other.msgs_),
-          args_tuple_(other.args_tuple_) {}
+          args_tuple_(other.args_tuple_),
+          main_thread_(other.main_thread_),
```

Comment:
So every `recursive_reducer` is a copy of the original `recursive_reducer` so the `main_thread_` is always set to the thread that made the first copy. At least, it is always one thread which is all you need to guarantee no race condition?

Reply:
Yes. Whatever reducer gets to execute in the main thread will get to use the outer AD tape which originated in that thread. Doing it by thread id ensures good thread locality, since the TBB does not guarantee anything where things run.